xc/xc7/tests/soc/ibex/ibex.patch

```diff
@@ -0,0 +1,110 @@
+diff --git a/examples/fpga/artya7/rtl/top_artya7.sv b/examples/fpga/artya7/rtl/top_artya7.sv
+index c26ca45..be19828 100644
+--- a/examples/fpga/artya7/rtl/top_artya7.sv
++++ b/examples/fpga/artya7/rtl/top_artya7.sv
+@@ -11,7 +11,7 @@ module top_artya7 (
+   parameter int          MEM_SIZE     = 64 * 1024; // 64 kB
+   parameter logic [31:0] MEM_START    = 32'h00000000;
+   parameter logic [31:0] MEM_MASK     = MEM_SIZE-1;
+-  parameter              SRAMInitFile = "";
++  parameter              SRAMInitFile = "led.vmem";
+ 
+   logic clk_sys, rst_sys_n;
+ 
+diff --git a/shared/rtl/fpga/xilinx/clkgen_xil7series.sv b/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
+index e41b4b0..e080de7 100644
+--- a/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
++++ b/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
+@@ -27,9 +27,9 @@ module clkgen_xil7series (
+     .STARTUP_WAIT         ("FALSE"),
+     .DIVCLK_DIVIDE        (1),
+     .CLKFBOUT_MULT        (12),
+-    .CLKFBOUT_PHASE       (0.000),
++    .CLKFBOUT_PHASE       (0),
+     .CLKOUT0_DIVIDE       (24),
+-    .CLKOUT0_PHASE        (0.000),
++    .CLKOUT0_PHASE        (0),
+     .CLKOUT0_DUTY_CYCLE   (0.500)
+   ) pll (
+     .CLKFBOUT            (clk_fb_unbuf),
+@@ -40,7 +40,7 @@ module clkgen_xil7series (
+     .CLKOUT4             (),
+     .CLKOUT5             (),
+      // Input clock control
+-    .CLKFBIN             (clk_fb_buf),
++    .CLKFBIN             (clk_fb_unbuf),
+     .CLKIN1              (io_clk_buf),
+     .CLKIN2              (1'b0),
+      // Tied to always select the primary input clock
+@@ -59,12 +59,6 @@ module clkgen_xil7series (
+     // Do not reset PLL on external reset, otherwise ILA disconnects at a reset
+     .RST                 (1'b0));
+ 
+-  // output buffering
+-  BUFG clk_fb_bufg (
+-    .I (clk_fb_unbuf),
+-    .O (clk_fb_buf)
+-  );
+-
+   BUFG clk_50_bufg (
+     .I (clk_50_unbuf),
+     .O (clk_50_buf)
+diff --git a/vendor/lowrisc_ip/prim/rtl/prim_lfsr.sv b/vendor/lowrisc_ip/prim/rtl/prim_lfsr.sv
+index cf35642..3b60d04 100644
+--- a/vendor/lowrisc_ip/prim/rtl/prim_lfsr.sv
++++ b/vendor/lowrisc_ip/prim/rtl/prim_lfsr.sv
+@@ -299,7 +299,7 @@ module prim_lfsr #(
+   ////////////////
+   // Galois XOR //
+   ////////////////
+-  if (64'(LfsrType) == 64'("GAL_XOR")) begin : gen_gal_xor
```

Comment:
Is this something sv2v doesn't handle?

Reply:
Yes, those variable casting creates strange wrappers in the final Verilog file, which cannot be read by the Yosys:
```
/mnt/hdd1/arch_defs_ws/ibex_PR/build/xc/xc7/tests/soc/ibex/ibex_sv2v.v:582: ERROR: Condition for generate if is not constant!
```

Reply:
This should be in the burndown list, and identify if this is yosys bug or a sv2v bug.

Reply:
This is a missing Yosys functionality. It seems to be added to the Yosys upstream, since the latest Yosys can parse the sv2v output without the patch.